src/actions/filterActions.js

```diff
@@ -0,0 +1,122 @@
+import queryString from 'query-string'
+import deepEquals from 'deep-equal'
+
+import * as eventActions from './eventActions'
+
+export const CHANGE_EVENT_FILTER = 'CHANGE_EVENT_FILTER'
+
+/*SHAPE OF FILTERS
+filters = {
+    incidentId: 0,
+    eventTypes: [{id: 0, name: testName}, ...]
+    occurredStart: someDateTime,
+    occurredEnd: someDateTime
+}*/
+
+export const changeEventFilter = (history) => (filter) => {
+    getUrlFromFilter(history, filter)
+    return {
+        type: CHANGE_EVENT_FILTER,
+        filter
+    }
+}
+
+export const addFilter = (history) => (filter, eventType) => (dispatch) => {
+    let newFilter = {}
+    let oldFilter = filter
+    if (!eventType && !eventType.id) {
+        return
+    }
+    if (oldFilter && oldFilter.eventTypes && oldFilter.eventTypes.map(eventType => eventType.id).includes(eventType.id)) {
+        newFilter = {...oldFilter}
+    }
+    else {
+        newFilter = {
+            ...oldFilter,
+            eventTypes: oldFilter.eventTypes ?
+            oldFilter.eventTypes.concat({
+                                    id: eventType.id,
+                                    name: eventType.name
+                                    })
+            : [{ id: eventType.id, name: eventType.name }]
+        }
+    }
+    dispatch(applyFilter(history)(oldFilter, newFilter))
+}
+
+export const removeFilter = (history) => (oldFilter, eventTypeToDelete) => (dispatch) => {
+    if (!oldFilter.eventTypes.map(eventType => eventType.id).includes(eventTypeToDelete.id)) {
+        return
+    }
+    const newFilter = {
+        ...oldFilter,
+        eventTypes: oldFilter.eventTypes.filter(eventType => eventTypeToDelete.id !== eventType.id)
+    }
+    dispatch(applyFilter(history)(oldFilter, newFilter))
+}
+
+export const applyFilter = (history) => (oldFilter, newFilter) => (dispatch) => {
+    if (!newFilter.incidentId) {
+        throw 'Need to filter on incidentId!'
+    }
+    if (!deepEquals(oldFilter, newFilter)) {
+        dispatch(changeEventFilter(history)(newFilter))
+        dispatch(eventActions.fetchEvents(newFilter))
+    }
+}
+
+export const synchronizeFilters = (filter, incidentId, ticketId, history, dispatch) => {
```

Comment:
Since this is an action, it should be called like:

dispatch(filterActions.synchronizeFilters(myFilter, myIncidentId, myTicketId, myHistory)

rather than

filterActions.synchronizeFilters(myFilter, myIncidentId, myTicketId, myHistory, dispatch)

the signature can change to 
export const synchronizeFilters = (filter, incidentId, ticketId, history) => {

then just return applyFilter(history)(filter, newFilter) instead of dispatching it.

Reply:
Fixed